### DISCUSSION TITLE
Is it worth updating the docs and samples to use the new "canonical" form of logback.xml?

### DISCUSSION THREAD

**Discussion by slisaasquatch** [playframework/playframework#11472] [Ideas]
Since Logback 1.3/1.4, they introduced a new canonical form of logback.xml while being backwards compatible with the legacy form. And they have a translator [here](https://logback.qos.ch/translator/services/xml2Canon.html).

It may seem trivial to use that translator tool to convert the files, but it seems like it gets rid of some of the custom stuff like this

```xml
<conversionRule conversionWord="coloredLevel" converterClass="play.api.libs.logback.ColoredLevel" />
```

**Comment by slisaasquatch** [comment]
I thought it was already on 2.18.17 but apparently not

**Reply by mkurz** [reply]
No, we did not upgrade the dependency in 2.8.17. But we made code changes so that people can upgrade it themselves without Play throwing exceptions.

**Comment by mkurz** [comment]
We can do that in the main branch and in the 2.9.x branch of play-samples: https://github.com/playframework/play-samples/tree/2.9.x

**Comment by wsargent** [comment]
`ColoredLevel` could be replaced with the built-in `highlight` https://logback.qos.ch/manual/layouts.html#coloring

**Comment by mkurz** [comment]
Got rid of `ColoredLevel` by replacing it with `highlight` in all logback configs and deprecated it. Also upgraded all logback configs to the new canonical format.

* https://github.com/playframework/playframework/pull/11662
* https://github.com/playframework/play-samples/pull/195
* https://github.com/playframework/play-socket.io/pull/260
* https://github.com/playframework/play-slick/pull/717
* https://github.com/playframework/play-webgoat/pull/88
* https://github.com/playframework/play-java-seed.g8/pull/128
* https://github.com/playframework/play-scala-seed.g8/pull/187
* https://github.com/playframework/play-webgoat/pull/89
* https://github.com/playframework/play-java-seed.g8/pull/127
* https://github.com/playframework/play-samples/pull/194
* https://github.com/playframework/play-scala-seed.g8/pull/186 
* https://github.com/playframework/play-json/pull/851
* https://github.com/playframework/play-samples/pull/196
* https://github.com/playframework/scalatestplus-play/pull/398
* https://github.com/playframework/play-webgoat/pull/90
* https://github.com/playframework/play-ws/pull/736
* https://github.com/playframework/play-socket.io/pull/262
* https://github.com/playframework/play-soap/pull/365
* https://github.com/playframework/play-slick/pull/719
* https://github.com/playframework/play-scala-seed.g8/pull/188
* https://github.com/playframework/play-java-seed.g8/pull/129
* https://github.com/playframework/anorm/pull/532
* https://github.com/playframework/playframework/pull/11659
